src/screencast/pipewire_screencast.c

```diff
@@ -170,28 +172,59 @@ static void pwr_handle_stream_add_buffer(void *data, struct pw_buffer *buffer) {
 	d = buffer->buffer->datas;
 
 	// Select buffer type from negotiation result
-	if ((d[0].type & (1u << SPA_DATA_MemPtr)) > 0) {
-		d[0].type = SPA_DATA_MemPtr;
+	if ((d[0].type & (1u << SPA_DATA_MemFd)) > 0) {
+		d[0].type = SPA_DATA_MemFd;
 	} else {
 		logprint(ERROR, "pipewire: unsupported buffer type");
 		cast->err = 1;
 		return;
 	}
 
+	logprint(TRACE, "pipewire: selected buffertype %d", d[0].type);
 	// Prepare buffer for choosen type
-	if (d[0].type == SPA_DATA_MemPtr) {
+	if (d[0].type == SPA_DATA_MemFd) {
+		d[0].type = SPA_DATA_MemFd;
 		d[0].maxsize = cast->simple_frame.size;
 		d[0].mapoffset = 0;
 		d[0].chunk->size = cast->simple_frame.size;
 		d[0].chunk->stride = cast->simple_frame.stride;
 		d[0].chunk->offset = 0;
-		d[0].flags = 0;
-		d[0].fd = -1;
+		d[0].flags = SPA_DATA_FLAG_READWRITE;
+		d[0].fd = anonymous_shm_open();
+
+		if (d[0].fd == -1) {
+			logprint(ERROR, "pipewire: unable to create anonymous filedescriptor");
+			return;
+		}
+
+		if (ftruncate(d[0].fd, d[0].maxsize) < 0) {
+			logprint(ERROR, "pipewire: unable to truncate filedescriptor");
```

Comment:
We should `close` the fd and reset it to -1 here

Reply:
done